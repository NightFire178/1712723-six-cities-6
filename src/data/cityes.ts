const Cityes = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`,
];
export default Cityes
