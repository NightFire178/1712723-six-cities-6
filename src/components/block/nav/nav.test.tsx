import React from 'react'
import {render} from '@testing-library/react'
import Nav from "./nav";
import cityes from "../../../data/cityes";

const appStateSelection = {
  cityNow(): string {
    return cityes[0]
  }
}
const useDispatch = ()=>(jest.fn())
jest.mock("../../../redux/selectors/app-state", ()=>appStateSelection )

jest.mock(`react-redux`, ()=>({useDispatch}))
test(`nav block component test`, () => {
  const header = render(<Nav/>)
  expect(header).toMatchSnapshot()
})
