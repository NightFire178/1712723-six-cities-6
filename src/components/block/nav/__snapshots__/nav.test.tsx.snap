// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav block component test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tabs"
      >
        <section
          class="locations container"
        >
          <ul
            class="locations__list tabs__list"
          >
            <a>
              <li
                class="locations__item"
              >
                <div
                  class="locations__item-link tabs__item tabs__item--active"
                >
                  <span>
                    Paris
                  </span>
                </div>
              </li>
            </a>
            <a>
              <li
                class="locations__item"
              >
                <div
                  class="locations__item-link tabs__item false"
                >
                  <span>
                    Cologne
                  </span>
                </div>
              </li>
            </a>
            <a>
              <li
                class="locations__item"
              >
                <div
                  class="locations__item-link tabs__item false"
                >
                  <span>
                    Brussels
                  </span>
                </div>
              </li>
            </a>
            <a>
              <li
                class="locations__item"
              >
                <div
                  class="locations__item-link tabs__item false"
                >
                  <span>
                    Amsterdam
                  </span>
                </div>
              </li>
            </a>
            <a>
              <li
                class="locations__item"
              >
                <div
                  class="locations__item-link tabs__item false"
                >
                  <span>
                    Hamburg
                  </span>
                </div>
              </li>
            </a>
            <a>
              <li
                class="locations__item"
              >
                <div
                  class="locations__item-link tabs__item false"
                >
                  <span>
                    Dusseldorf
                  </span>
                </div>
              </li>
            </a>
          </ul>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tabs"
    >
      <section
        class="locations container"
      >
        <ul
          class="locations__list tabs__list"
        >
          <a>
            <li
              class="locations__item"
            >
              <div
                class="locations__item-link tabs__item tabs__item--active"
              >
                <span>
                  Paris
                </span>
              </div>
            </li>
          </a>
          <a>
            <li
              class="locations__item"
            >
              <div
                class="locations__item-link tabs__item false"
              >
                <span>
                  Cologne
                </span>
              </div>
            </li>
          </a>
          <a>
            <li
              class="locations__item"
            >
              <div
                class="locations__item-link tabs__item false"
              >
                <span>
                  Brussels
                </span>
              </div>
            </li>
          </a>
          <a>
            <li
              class="locations__item"
            >
              <div
                class="locations__item-link tabs__item false"
              >
                <span>
                  Amsterdam
                </span>
              </div>
            </li>
          </a>
          <a>
            <li
              class="locations__item"
            >
              <div
                class="locations__item-link tabs__item false"
              >
                <span>
                  Hamburg
                </span>
              </div>
            </li>
          </a>
          <a>
            <li
              class="locations__item"
            >
              <div
                class="locations__item-link tabs__item false"
              >
                <span>
                  Dusseldorf
                </span>
              </div>
            </li>
          </a>
        </ul>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
